From 09310b25d96aa76445e1de4159384954415769d6 Mon Sep 17 00:00:00 2001
From: Micha LaQua <micha.laqua@gmail.com>
Date: Thu, 16 Jan 2014 15:44:52 +0100
Subject: [PATCH] mm-video-legacy: vdec: display: Change in screenRefresh
 interface

Display interface changed to optimise binder transaction.

* backported from media-caf (ChangeID I7f7861e2626fdd32efcfe3a2575a62caed91f8db)

Change-Id: I602e2eb82dfa5304bce478a102ca2f7cadd05856
---
 mm-video-legacy/vidc/vdec/inc/omx_vdec.h   |  1 -
 mm-video-legacy/vidc/vdec/src/omx_vdec.cpp | 26 +++++++++-----------------
 2 files changed, 9 insertions(+), 18 deletions(-)

diff --git a/mm-video-legacy/vidc/vdec/inc/omx_vdec.h b/mm-video-legacy/vidc/vdec/inc/omx_vdec.h
index a201fb5..10853b0 100644
--- a/mm-video-legacy/vidc/vdec/inc/omx_vdec.h
+++ b/mm-video-legacy/vidc/vdec/inc/omx_vdec.h
@@ -57,7 +57,6 @@ static ptrdiff_t x;
 #endif
 #include <binder/MemoryHeapBase.h>
 #include <ui/ANativeObjectBase.h>
-#include <binder/IServiceManager.h>
 extern "C"{
 #include<utils/Log.h>
 }
diff --git a/mm-video-legacy/vidc/vdec/src/omx_vdec.cpp b/mm-video-legacy/vidc/vdec/src/omx_vdec.cpp
index 45805c0..d6ce0b9 100644
--- a/mm-video-legacy/vidc/vdec/src/omx_vdec.cpp
+++ b/mm-video-legacy/vidc/vdec/src/omx_vdec.cpp
@@ -49,6 +49,7 @@ ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <fcntl.h>
 #include <limits.h>
 #include <qdMetaData.h>
+#include <QServiceUtils.h>
 
 #ifndef _ANDROID_
 #include <sys/ioctl.h>
@@ -9375,18 +9376,13 @@ int omx_vdec::secureDisplay(int mode) {
     }
 
     sp<IServiceManager> sm = defaultServiceManager();
-    sp<qService::IQService> displayBinder =
-        interface_cast<qService::IQService>(sm->getService(String16("display.qservice")));
 
-    if (displayBinder != NULL) {
-        displayBinder->securing(mode);
+    if (m_secure_display == false) {
+        securing(mode);
         if (mode == qService::IQService::END) {
             m_secure_display = true;
         }
     }
-    else {
-        DEBUG_PRINT_ERROR("secureDisplay(%d) display.qservice unavailable", mode);
-    }
     return 0;
 }
 
@@ -9395,18 +9391,14 @@ int omx_vdec::unsecureDisplay(int mode) {
         return 0;
     }
 
-    if (mode == qService::IQService::END) {
-        m_secure_display = false;
-    }
-
     sp<IServiceManager> sm = defaultServiceManager();
-    sp<qService::IQService> displayBinder =
-        interface_cast<qService::IQService>(sm->getService(String16("display.qservice")));
 
-    if (displayBinder != NULL)
-        displayBinder->unsecuring(mode);
-    else
-        DEBUG_PRINT_ERROR("unsecureDisplay(%d) display.qservice unavailable", mode);
+    if (m_secure_display == true) {
+        unsecuring(mode);
+        if (mode == qService::IQService::END) {
+            m_secure_display = false;
+        }
+    }
     return 0;
 }
 
-- 
1.8.5

